#
# patch fixes CR 24747309 (passive mode should treat EACCES as EADDRINUSE)
# fix has been sent to upstream master branch:
#	https://github.com/proftpd/proftpd/pull/703
#

diff --git a/src/inet.c b/src/inet.c
index b01230d3e..da26360a8 100644
--- a/src/inet.c
+++ b/src/inet.c
@@ -433,7 +433,11 @@ static conn_t *init_conn(pool *p, int fd, const pr_netaddr_t *bind_addr,
       }
 
       if (res != -1 ||
+#ifdef	SOLARIS2
+          (hold_errno != EADDRINUSE && hold_errno != EACCES) ||
+#else	/* !SOLARIS2 */
           hold_errno != EADDRINUSE ||
+#endif	/* SOLARIS2 */
           (port != INPORT_ANY && !retry_bind)) {
         break;
       }
